### ISSUE TITLE
3.9.0 can possibly broke build

### ISSUE THREAD

**Issue by dutsky** [widgetbook/widgetbook#1266]
**Description**
If you have previous widgetbook version with caret, like `widgetbook: ^3.7.0`, and you have old flutter, your build may be broken if you pubget today.

This happens because 3.9.0 relies on new accessibilty_tools 2.0.0.

**Steps To Reproduce**

1. Stick with old flutter
2. Have `widgetbook: ^3.7.0`
3. Pub get after tonight :)
4. See build error like 
```
../../../../../.pub-cache/hosted/pub.dev/accessibility_tools-2.2.1/lib/src/accessibility_tools.dart:386:28: Error: The getter 'globalPosition' isn't defined for the class 'DragEndDetails'.
 - 'DragEndDetails' is from 'package:flutter/src/gestures/drag_details.dart' ('../../../../../fvm/versions/3.16.9/packages/flutter/lib/src/gestures/drag_details.dart').
Try correcting the name to the name of an existing getter, or defining a getter or field named 'globalPosition'.
    _panPosition = details.globalPosition;
                           ^^^^^^^^^^^^^^
Failed to compile application.
```

**Expected Behavior**
New versions with breaking changes must increase major version, not minor

**Screenshots**

**Additional Context**

**Comment by yousinix** [comment]
Hello @dutsky 👋 
Thanks for letting us know about this!

We have updated `accessibility_tools` to `^2.0.0` because [our minimum Flutter version](https://github.com/widgetbook/widgetbook/blob/d318a008c60856f2387f3f85244e5b1e3173650b/packages/widgetbook/pubspec.yaml#L16) now is 3.16, which [matches](https://github.com/rebelappstudio/accessibility_tools/blob/6184d07ee0d635ed10f7291eb951f41d7b93d415/pubspec.yaml#L8) the one from that package.

We are not in favor of upgrading Flutter versions in minor releases, but sadly that's what have been happening in the Dart ecosystem for a while. To prevent this from happening in the future  _(starting from Widgetbook v4)_, we are planning on having zero dependencies, that will not [force us](https://github.com/widgetbook/widgetbook/issues/1197#issuecomment-2162774137) to update. 

Assuming you have `widgetbook: ^3.7.0` in your `pubspec.yaml`, then my expectation is that should not break your builds because:

1. If you have Flutter version <3.16, then `pub get` will not get v3.9.0 as it doesn't match the Flutter version constraint.
1. If you have Flutter version >=3.16, then `pub get` will update `widgetbook` to `3.9.0` and `accessibility_tools` to `^2.0.0`, where they should be both compatible with your Flutter version.

If that's not the case for you, please provide as with a reproduction repo that can possibly break. Make sure to check-in `pubspec.lock` to your repo, and specifying the Flutter version.

**Comment by dutsky** [comment]
Thank you!
I'm sorry. You can:
0. have flutter 3.16.9
1. create a brand new flutter project
2. set `sdk: '>=3.0.0 <4.0.0'`
3. do `flutter pub add widgetbook:3.9.0`
4. in your code add `Widgetbook.material(directories: [])` widget somewhere
5. run or build project
you can get error like this:
```
../../../.pub-cache/hosted/pub.dev/accessibility_tools-2.2.1/lib/src/accessibility_tools.dart:386:28: Error: The getter 'globalPosition' isn't defined for the class 'DragEndDetails'.
 - 'DragEndDetails' is from 'package:flutter/src/gestures/drag_details.dart' ('../../../fvm/versions/3.16.9/packages/flutter/lib/src/gestures/drag_details.dart').
Try correcting the name to the name of an existing getter, or defining a getter or field named 'globalPosition'.
    _panPosition = details.globalPosition;
                           ^^^^^^^^^^^^^^
Failed to compile application.
```

**Comment by yousinix** [comment]
The problem lies in `accessibility_tools-2.2.1` where they have [minimum Flutter version set to 3.16](https://github.com/rebelappstudio/accessibility_tools/blob/7a793c058f3aae5b31503d2f0156615f6a269e5c/pubspec.yaml#L12), which does not work properly (rebelappstudio/accessibility_tools#52).

They patched it in version 2.2.2. Ideally, if all dependencies has proper constraints, then Widgetbook 3.9.0 should not break. To prevent this in the future, we will aim to reduce our dependencies (as I have mentioned earlier), to provide us with ultimate control over our own package.

To fix this, please consider one of the following:
1. Using [`dependency_overrides`](https://dart.dev/tools/pub/dependencies#dependency-overrides) to set `accessibility_tools` to version 2.1.0 or lower.
1. Upgrading your Flutter to be `3.22.0` if possible.


In the meantime, I will be closing this, as there is nothing we can do from our side.
Thanks for bringing this up to our attention!
